File: 이우엽/Week_15/P43165_타겟넘버.java
class Solution {
    int result = 0;

    public void DFS(int[] numbers, int target, int index, int sum) {
        if(index == numbers.length) {
            if(sum == target) {
                result++;
            }
            return;
        }

        DFS(numbers, target, index+1, sum+numbers[index]);
        DFS(numbers, target, index+1, sum-numbers[index]);
    }

    public int solution(int[] numbers, int target) {
        int answer = 0;
        DFS(numbers, target, 0, 0);

        answer = result;
        return answer;
    }
}

File: 윤상우/Week_15/P43165_타겟넘버.py
def solution(numbers, target):
    return dfs(numbers, target, 0, 0)
    
def dfs(numbers, target, idx, sum):
    if idx == len(numbers):
        return 1 if sum == target else 0

    ways = 0
    ways += dfs(numbers, target, idx + 1, sum + numbers[idx])
    ways += dfs(numbers, target, idx + 1, sum - numbers[idx])
    return ways